packages/phpstan-rules/docs/rules_overview.md

```diff
@@ -338,6 +338,98 @@ class SomeCommand extends Command
 
 <br>
 
+## ClassNamespaceGuardRule
+
+Define in which namespaces (using *, ** or ? glob-like pattern matching) can classes extending specified class or implementing specified interface exist
```

Comment:
I think the rule name could explain the definition in more clear way. The "guard" is already included in the rule, as they usually forbid some behavior, so we can remove it. What about including "eclusive" and "extending" keywords?

Reply:
How about `ClassExtendingExclusiveNamespaceRule`? Its really `extends` or `implements`, but for the simplicity sake?

Reply:
Sounds good :+1: